src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/SetOperation.java

```diff
@@ -31,8 +31,10 @@ public static SetOperation construct(ObjectNode args) {
   public boolean updateDocument(ObjectNode doc, UpdateTargetLocator targetLocator) {
     boolean modified = false;
     for (SetAction addition : additions) {
-      JsonNode prev = doc.replace(addition.path, addition.value);
-      modified |= !Objects.equals(prev, addition.value);
+      UpdateTarget target = targetLocator.findOrCreate(doc, addition.path());
+      JsonNode newValue = addition.value();
+      JsonNode oldValue = target.replaceValue(newValue);
+      modified |= !Objects.equals(newValue, oldValue);
```

Comment:
Does equals handles the order of element? if not need to check the order of the elements incase the set done for object type?

Reply:
`JsonNode.equals()` does not consider order, true. But individual `$set` operation never changes order of existing properties so I think this is still safe. To change order, I think removal plus addition (or addition of non-existing) property is needed.

Reply:
Actually, I think I should refactor things slightly, to extract comparison method just in case this needs to be reconsidered -- I will create an issue.
I think I see the potential problem: replacing an existing JSON Object but with different field order.

Reply:
Created #190 for future follow-up work.